### PR TITLE
add nodeport-proxy

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This copies the NodePort proxy code from https://github.com/kubermatic/nodeport-proxy.
I'll issue a second PR which will update the helm chart once this got merged.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/cmd/nodeport-proxy/Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+docker: build
+	docker build -t quay.io/kubermatic/nodeport-proxy:$(TAG) .
+	docker push quay.io/kubermatic/nodeport-proxy:$(TAG)
+
+test:
+	go test ./...
+
+lint:
+	golangci-lint run -E gosec -E interfacer -E goconst -E gofmt -E goimports -E misspell -E unparam -E prealloc ./...
```
Comment (alvaroaleman) [inline api/cmd/nodeport-proxy/Makefile]: The `lint` target in the `api` folder already covers this code, right?

Comment (mrIncompetent) [reply api/cmd/nodeport-proxy/Makefile]: Removed with https://github.com/kubermatic/kubermatic/pull/2872/commits/a185cef952ec4e13a174a09dfaab97e85bc71d65

**Pull request merged**